### ISSUE TITLE
500 Internal Server Error on registry  http://registry.npmjs.org/

### ISSUE THREAD

**Issue by mkantautas** [npm/npm#20962]
#### I'm opening this issue because:

  - [x ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] npm is producing incorrect or undesirable behavior.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
Can't use ````npm install```` or upgrade it. 
My problem can be well illustrated with
````curl http://registry.npmjs.org/````  which returns :  
````
<html>
<head>
<title>Fastly error: unknown domain</title>
</head>
<body>
Fastly error: unknown domain. Please check that this domain has been added to a service.
</body>
</html>
````
On the other hand my ````curl http://registry.npmjs.com/```` works, but if I use this as my registry, I still get error from few child dependencies that try to reach http://registry.npmjs.org/ - and get 500.
My ````/etc/hosts```` file is clean(curl http://registry.npmjs.org/ is commented out) and dnsmq plugin is commented out it network config...
#### How can the CLI team reproduce the problem?

https://gist.github.com/neorganic/cfb796160eb462e5f378b343dc80978f

### supporting information:

 - `npm -v` prints: 6.0.0
 - `node -v` prints: v8.2.1
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux Ubuntu 16.04 lts
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [x ] I develop using Vagrant on OS X or Linux.
   - [x ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by legodude17** [comment]
Does this occur if you first do `npm config set registry https://registry.npmjs.org`?

**Comment by mkantautas** [comment]
@legodude17  Yes, that doesn't affect the outcome

**Comment by legodude17** [comment]
What does `npm ping` do? Does `ping http://registry.npmjs.org` work?

**Comment by mkantautas** [comment]
````
ping http://registry.npmjs.org
ping: unknown host http://registry.npmjs.org
````

**Comment by legodude17** [comment]
That's odd. Does opening http://registry.npmjs.org in a browser work?

**Comment by mkantautas** [comment]
No same issue:  
````Fastly error: unknown domain. Please check that this domain has been added to a service.````

**Comment by legodude17** [comment]
Interesting. It works fine for me. I suggest you report this to https://github.com/npm/registry/issues, as it seems to be a registry issue.

**Comment by mkantautas** [comment]
I think something is bad with my device on some level - my mobile from the same network can access that page. Will try to reinstall/upgrade my distro and try again ;)

**Comment by mkantautas** [comment]
Resolved with ````nameserver 8.8.8.8```` added to ````/etc/resolv.conf````

**Comment by mkantautas** [comment]
^ This was just a very hacky temp workaround. This issue was the last drop - that made me move to docker. Now everything is rock solid! ;)
